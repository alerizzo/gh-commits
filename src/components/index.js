export { default as Skeleton } from './Skeleton/Skeleton';

export { default as TextInput } from './TextInput/TextInput';
export { default as Button } from './Button/Button';

export { default as Layout } from './Layout/Layout';

export { default as CommitsList } from './CommitsList/CommitsList';
export { default as RepositorySearch } from './RepositorySearch/RepositorySearch';
