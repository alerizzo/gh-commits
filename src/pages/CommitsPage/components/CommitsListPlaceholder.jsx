import React from 'react';
import { CommitsList } from 'components';

function CommitsListPlaceholder({ message, loading }) {
  return (
    <div className="CommitsListPlaceholder">
      <CommitsList.Skeleton rows={3} loading={loading} />
      {message && <div className="overlay is-size-5 has-text-weight-bold">{message}</div>}
    </div>
  );
}

export default CommitsListPlaceholder;
