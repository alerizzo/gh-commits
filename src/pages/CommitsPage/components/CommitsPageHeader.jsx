import React from 'react';
import { TextInput } from 'components';
import CommitRow from 'components/CommitsList/CommitRow';

export default function CommitsPageHeader({ onSearch, repository, disabled, showTableHeader }) {
  return (
    <div className="CommitsPageHeader">
      <h1 className="title is-3">Commits list</h1>
      <div className="field">
        <label className="label is-size-6">
          Search
          {repository && (
            <span className="has-text-weight-light">
              {' in '}
              {repository.owner}'s {repository.repository} ({repository.refName})
            </span>
          )}
        </label>
        <TextInput style={{ maxWidth: '50%' }} disabled={disabled} placeholder="Search ..." />
      </div>
      {showTableHeader && (
        <div className="CommitsPageHeader_TableHeader">
          <CommitRow.Header />
        </div>
      )}
    </div>
  );
}
