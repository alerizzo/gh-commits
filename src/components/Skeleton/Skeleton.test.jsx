import React from 'react';
import renderer from 'react-test-renderer';

import { Skeleton } from 'components';

describe('<Skeleton.Circle />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Skeleton.Circle {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Skeleton.Text />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Skeleton.Text {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
