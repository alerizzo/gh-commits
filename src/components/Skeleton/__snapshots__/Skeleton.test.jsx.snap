// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton.Circle /> render 1`] = `
<div
  className="Skeleton Skeleton_Circle "
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`<Skeleton.Text /> render 1`] = `
<div
  className="Skeleton Skeleton_Text "
  style={
    Object {
      "width": undefined,
    }
  }
/>
`;
